### ISSUE TITLE
Discussion around renaming Node to Client

### ISSUE THREAD

**Issue by cf19drofxots** [counterfactual/monorepo#1522]
There has been discussion around renaming the Node to Client.

The reason why I'm not personally for this renaming is because the functionality provided by this package is not restricted to traditional client functionality, and depending on the context of operation, it could act as a server or simply a peer (intermediary) relaying some info.

Other thoughts around this are welcome cc @snario @ebryn @ArjunBhuptani @rhlsthrm etc

**Comment by ebryn** [comment]
Yeah I don't think it makes sense personally. Where'd this discussion begin? I'd like to see more context.

**Comment by snario** [comment]
At the state channels workshop we just said abstractly that a better name for this piece of the tech stack is "Client". Then I said that since we have a class called "Node" maybe that class should be called "Client"?

It could be a dumb suggestion but thought I'd mention it.

It seems the trend for these things is to either name it `<language abbreviation>-<protocol>` or something cute like "eclair". These are the names of the Lightning clients:

1. Elements Project – c-lightning (C)
2. Acinq – eclair (Scala)
3. Lightning Labs – Lightning Network Daemon (Go)

**Comment by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll** [comment]
Yeah, my view is that "node" is too generic a name, and "client" as well. Something like `lnd` points to a particular implementation while clearly excluding certain functionality (e.g. wallet UI in lightning's case)

**Comment by rhlsthrm** [comment]
I like the idea of a non-related name like the `eclair` example, otherwise you end up with these arguments about if the name matches the functionality. `node-provider` would definitely have to change with whatever this changes to, I'd imagine.
